andela/zinnia-ah-backend
# #164797165 capture and show user reading stats

#### What does this PR do?
Keeps track of the user reading activities and display them to an authenticated user

#### Description of Task to be completed?
- [x] keep track of article view clicks
- [x] return stats to the authenticated user

#### How should this be manually tested?
- Clone the repo,
- Run ```npm install``` to install all dependencies,
- Create a .env file using the .env.example as a guide and enter your environment variables
- Ensure you have a PostgreSQL server running on your development environment
- Run `sequelize db:create`  
- Run `sequelize db:migrate` 
- Run `sequelize db:seed:all`   
- Run npm run start:dev,
- On Postman, send a POST request `localhost:3000/api/v1/auth/login` providing `igbominadeveloper@ah.com` as `email` and `favourafolayan` as `password`. Take the token provided and set it in the header authorization field.
- On Postman, send a GET request to`localhost:3000/api/v1/articles/:articleSlug`
- On Postman, send a GET request `localhost:3000/api/v1/users/profiles/igbominadeveloper/stats` 
- You should get the screenshot below
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories? (if applicable)
[#164797165](https://www.pivotaltracker.com/story/show/164797165)

#### Screenshots
<img width="1440" alt="Screenshot 2019-04-23 at 11 19 05 AM" src="https://user-images.githubusercontent.com/12516195/56573788-c2056780-65b9-11e9-95d2-e94553c65569.png">
